### PR TITLE
IMap local values

### PR DESCRIPTION
**Pull request by LarsKrogJensen**
This PR provides support IMap.localValues and IMap.localValues(Predicate) to provide fast access to locally own values.

Changes are few and trivial 
 
Breaking changes (list specific methods/types/messages):
* API
* client protocol format
* serialized form
* snapshot format

Checklist:
- [ ] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [ ] Label `Add to Release Notes` or `Not Release Notes content` set
- [ ] Request reviewers if possible
- [ ] Send backports/forwardports if fix needs to be applied to past/future releases
- [ ] New public APIs have `@Nonnull/@Nullable` annotations
- [ ] New public APIs have `@since` tags in Javadoc

### CODE REVIEW

**Review by srknzl — APPROVED** [review]
This is not really related to client side. If core approves, I approve for client team

**Review by TomaszGaweda — APPROVED** [review]
Very nice PR, thanks!

**Comment by LarsKrogJensen** [comment]
I made a start import cleanup, please re-run pr-builder

**Comment by LarsKrogJensen** [comment]
Also, but be good if this PR could be cherry picked into 5.3.x, I can provide a PR for that if needed

**Comment by srknzl** [comment]
run-lab-run

**Review by vbekiaris — APPROVED** [review]

File: hazelcast/src/main/java/com/hazelcast/map/IMap.java [hunk 1]
```diff
@@ -2455,6 +2455,47 @@ UUID addEntryListener(@Nonnull MapListener listener,
...
+     * The collection is <b>NOT</b> backed by the map,
+     * so changes to the map are <b>NOT</b> reflected in the collection.
+     * <p>
+     * This method is always executed by a distributed query,
+     * so it may throw a {@link QueryResultSizeExceededException}
+     * if {@link ClusterProperty#QUERY_RESULT_SIZE_LIMIT} is configured.
+     *
+     * @return an immutable collection clone of the values contained in this map
+     * @throws QueryResultSizeExceededException if query result size limit is exceeded
+     * @see ClusterProperty#QUERY_RESULT_SIZE_LIMIT
```
Comment (vbekiaris) [inline hazelcast/src/main/java/com/hazelcast/map/IMap.java:2472]: can you please add `@since 5.4.0` in the new methods' javadoc?

**Comment by vbekiaris** [comment]
thanks for the contribution @LarsKrogJensen . Once you add the `@since` javadoc we will merge this PR.

About backporting to 5.3 line: we have a policy of not introducing new public API in patch releases, so this will become available once 5.4.0 is released.

File: hazelcast/src/main/java/com/hazelcast/map/IMap.java (hunk 1 above)
Comment (pveentjer) [reply hazelcast/src/main/java/com/hazelcast/map/IMap.java:2472]: I'll deal with this in a follow up PR.

**Pull request merged**